From 719c17ec8175373fc15c9c8d4a057ffc7275fb36 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Tue, 17 Jan 2017 16:01:42 +0000
Subject: [PATCH 2192/2748] MAPREDUCE-6831. Flaky test
 TestJobImpl.testKilledDuringKillAbort.
 Contributed by Peter Bacsko

(cherry picked from commit 78b487bde175544ebe40e4dafab35569baa1d79e)

Change-Id: I729111c70c4852458b3298ab8e43cb5da3be702e
---
 .../mapreduce/v2/app/job/impl/TestJobImpl.java     |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TestJobImpl.java b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TestJobImpl.java
index f126f3d..4ddb803 100644
--- a/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TestJobImpl.java
+++ b/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/test/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TestJobImpl.java
@@ -488,9 +488,10 @@ public synchronized void abortJob(JobContext jobContext, State state)
   public void testKilledDuringKillAbort() throws Exception {
     Configuration conf = new Configuration();
     conf.set(MRJobConfig.MR_AM_STAGING_DIR, stagingDir);
+    // not initializing dispatcher to avoid potential race condition between
+    // the dispatcher thread & test thread - see MAPREDUCE-6831
     AsyncDispatcher dispatcher = new AsyncDispatcher();
-    dispatcher.init(conf);
-    dispatcher.start();
+
     OutputCommitter committer = new StubbedOutputCommitter() {
       @Override
       public synchronized void abortJob(JobContext jobContext, State state)
-- 
1.7.9.5

